esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2298,6 +2266,70 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
     return true;
 }
 
+void CFileSprayEx::checkDZScopeAccessAndSetSpraySourceDFUFileSpec(IEspContext &context, const char *srcPlane, const char *srcHost,
+    const char *srcFile, MemoryBuffer &srcXML, IDFUfileSpec *srcDFUfileSpec)
+{
+    if(srcXML.length() == 0)
+    {
+        //Both srcPlane and srcHost are validated in readAndCheckSpraySourceReq().
+        StringBuffer fnamebuf(srcFile);
+        fnamebuf.trim();
+        StringArray files;
+        files.appendList(fnamebuf, ","); // handles comma separated files
+        ForEachItemIn(i, files)
+        {
+            const char *file = files.item(i);
+            if (isEmptyString(file))
+                continue;
+
+            //Based on the tests, the dfuserver only supports the wildcard inside the file name, like '/path/f*'.
+            //The dfuserver throws an error if the wildcard is inside the path, like /p*ath/file.
+            StringBuffer dirPath, dirTail;
+            splitFilename(file, &dirPath, &dirPath, &dirTail, &dirTail);
+
+            const char* s = containsWildcard(dirTail) ? dirPath : file;
+            SecAccessFlags permission = getDropZoneScopePermissions(context, srcPlane, s, srcHost);
+            if (permission < SecAccess_Read)
+                throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Access DropZone Scope %s %s not allowed for user %s (permission:%s). Read Access Required.",
+                    srcPlane, s, context.queryUserId(), getSecAccessFlagName(permission));
+        }
+
+        SocketEndpoint ep(srcHost);
+        if (ep.isNull())
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Cannot resolve source network IP from %s.", srcHost);
+
+        RemoteMultiFilename rmfn;
```

Comment:
I see the change at the head of this function from parsing via RemoteMultiFilename to using a StringArray (related to my previous comment I think : https://github.com/hpcc-systems/HPCC-Platform/pull/17144#discussion_r1150512002).

But afaics, the code has become more complex rather than simpler, because it's now using both RemoteMultiFilename anStringArray, becuase setMultiFilename is still expecting a setMultiFilename.

It is not worth changing if it makes the code hard to interpret, if this multi filename semantic only needs to be a csv of local files, then it really should be using a simpler construct that RemoteMultiFilename (e.g. StringArray), but without follwing through and changing it so RemoteMultiFilename is not needed at all (i.e. by setMultiFilename), which by the looks of it, also only ever deals in local files, it is not worth making the change here.

Please don't change setMultiFilename at this time (it is beyond the scope of this PR).
Please revert the change made here to how it was, so it is as concise/clear as possible.
We can revisit the unnecessary use of RemoteMultiFilename at another time if we really want to.